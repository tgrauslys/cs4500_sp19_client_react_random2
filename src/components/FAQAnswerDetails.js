import React from 'react'
import FAQAnswerService from '../services/FAQAnswerService'
class FAQAnswerDetails extends React.Component {
    constructor(props) {
        super(props)
        this.faqAnswerService = FAQAnswerService.getInstance()
        this.state = {
            faqAnswers: [],
            faqAnswer: {
                choiceAnswer: '',
                id: 1
            }
        }
    }
    componentDidMount() {
        this.faqAnswerService
            .findAllFAQAnswers()
            .then(faqAnswers => {
                    this.props.history.push("/admin/faq-answers/" + faqAnswers[0].id)
                    this.setState({
                        faqAnswers: faqAnswers,
                        faqAnswer: faqAnswers[0]
                    })
                }
            )
    }
    selectFAQAnswer = id =>
        this.faqAnswerService
            .findFAQAnswerById(id)
            .then(faqAnswer => {
                    this.props.history.push("/admin/faq-answers/" + id)
                    this.setState({
                        faqAnswer: faqAnswer
                    })
                }
            )
    render() {
        return(
            <div>
                <h3>FAQ Answer Details</h3>
                <select
                    value={this.state.faqAnswer.id}
                    onChange={(e) => this.selectFAQAnswer(e.target.value)}
                    className="form-control">
                    {
                        this.state.faqAnswers
                            .map(faqAnswer =>
                                <option
                                    value={faqAnswer.id}
                                    key={faqAnswer.id}>
                                    {faqAnswer.id}
                                </option>
                            )
                    }
                </select>
                <label>FAQ Answer Question</label><br/>
                <input
                    onChange={() => {}}
                    className="form-control"
                    value={this.state.faqAnswer.question}/>
                <label>FAQ Answer Answer</label><br/>
                <input
                    onChange={() => {}}
                    className="form-control"
                    value={this.state.faqAnswer.answer}/>
            </div>
        )
    }
}

export default FAQAnswerDetails