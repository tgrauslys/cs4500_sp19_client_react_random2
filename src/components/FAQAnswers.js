import React from 'react'
import FAQAnswerService from '../services/FAQAnswerService'
class FAQAnswers extends React.Component {
    constructor(props) {
        super(props)
        this.faqAnswerService = FAQAnswerService.getInstance()
        this.state = {
            faqAnswers: []
        }
    }
    componentDidMount() {
        this.faqAnswerService
            .findAllFAQAnswers()
            .then(faqAnswers =>
                this.setState({
                    faqAnswers: faqAnswers
                })
            )
    }
    render() {
        return(
            <div>
                <h3>Service Answers</h3>
                <table className="table">
                    <tbody>
                    {
                        this.state.faqAnswers
                            .map(faqAnswer =>
                                <tr key={faqAnswer.id}>
                                    <td>{faqAnswer.question}</td>
                                    <td>{faqAnswer.answer}</td>
                                </tr>
                            )
                    }
                    </tbody>
                </table>
            </div>
        )
    }
}

export default FAQAnswers