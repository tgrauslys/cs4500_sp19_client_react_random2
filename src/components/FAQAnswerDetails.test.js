import React from 'react'
import FAQAnswerDetails from './FAQAnswerDetails'
import TestRenderer from 'react-test-renderer';
import FAQAnswers from '../data/faqanswers.mock'

test('Render FAQAnswer Details page', () => {
    const testRenderer = TestRenderer.create(
        <FAQAnswerDetails
            faqAnswers = {FAQAnswers}
            faqAnswer={FAQAnswers[0]}
            updateFormAnswer={null}
            navigateBack={null}
            editAnswer={null}
        />)

    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance = testRenderer.root
    const formControls = testInstance.findAllByProps({className: "form-control"});
    const delButton = testInstance.findByProps({class: "btn btn-warning btn-lg"});
    const updateButton = testInstance.findByProps({class: "btn btn-danger btn-lg active"})


    expect(formControls.length).toBe(3);
    expect(delButton).toBeDefined();
    expect(updateButton).toBeDefined();
    expect(formControls[1].props.value).toBe("What should the customer know about your pricing (e.g., discounts, fees)?");
    expect(formControls[2].props.value).toBe("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum");

})