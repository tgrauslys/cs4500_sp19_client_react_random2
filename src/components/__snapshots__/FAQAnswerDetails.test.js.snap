// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FAQAnswer Details page 1`] = `
<div>
  <h3>
    FAQ Answer Details
  </h3>
  <select
    className="form-control"
    value={1}
  >
    <option
      value={1}
    >
      1
    </option>
    <option
      value={2}
    >
      2
    </option>
  </select>
  <label>
    FAQ Answer Question
  </label>
  <br />
  <input
    className="form-control"
    value="What should the customer know about your pricing (e.g., discounts, fees)?"
  />
  <label>
    FAQ Answer Answer
  </label>
  <br />
  <input
    className="form-control"
    onChange={[Function]}
    value="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
  />
  <br />
  <td>
    <a
      class="btn btn-danger btn-lg active"
      onClick={[Function]}
      role="button"
    >
      Cancel
    </a>
  </td>
  <td>
    <a
      class="btn btn-warning btn-lg"
      onClick={[Function]}
      role="button"
    >
      Edit
    </a>
  </td>
</div>
`;
